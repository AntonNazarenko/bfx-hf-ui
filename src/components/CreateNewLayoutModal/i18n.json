{
    "ru-RU": {
        "addLayout": "Добавить Макет",
        "layoutName": "Имя Макета",
        "labelEmpty": "Поле Пусто"
    },
    "en-EN": {
        "addLayout": "Add Layout",
        "layoutName": "Layout Name",
        "label": "Label Empty"
    },
    "zh-ZH": {

    }
}