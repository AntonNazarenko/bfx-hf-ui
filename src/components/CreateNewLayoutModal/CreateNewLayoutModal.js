import React from 'react'
import _isEmpty from 'lodash/isEmpty'

import Input from '../../ui/Input'
import Modal from '../../ui/Modal'
import Button from '../../ui/Button'

import { propTypes, defaultProps } from './CreateNewLayoutModal.props'
import i18n from './i18n.json'
import './style.css'

const dictionary = i18n['ru-RU']
export default class CreateNewLayoutModal extends React.Component {
  static propTypes = propTypes
  static defaultProps = defaultProps

  state = {
    label: '',
    error: '',
  }

  constructor(props) {
    super(props)

    this.onLabelChange = this.onLabelChange.bind(this)
    this.onSubmit = this.onSubmit.bind(this)
  }

  onLabelChange(label) {
    this.setState(() => ({ label }))
  }

  onSubmit() {
    const { label } = this.state
    const { onSubmit, onClose } = this.props

    if (_isEmpty(label)) {
      this.setState(() => ({ error: dictionary.labelEmpty }))
      return
    }

    onSubmit(label)
    onClose()
  }

  render() {
    const { onClose } = this.props
    const { label, error } = this.state

    return (
      <Modal
        onClose={onClose}
        className='hfui-createnewlayoutmodal__wrapper'
        label={dictionary.addLayout}
        actions={(
          <Button
            green
            onClick={this.onSubmit}
            label={[
              <i key='icon' className='icon-plus' />,
              <p key='text'>{dictionary.addLayout}</p>,
            ]}
          />
        )}
      >
        <Input
          type='text'
          placeholder={dictionary.layoutName}
          value={label}
          onChange={this.onLabelChange}
        />

        {!_isEmpty(error) && (
          <p className='error'>{error}</p>
        )}
      </Modal>
    )
  }
}
