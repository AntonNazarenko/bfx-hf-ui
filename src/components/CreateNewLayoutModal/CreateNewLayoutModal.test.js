import React from 'react'
import { configure, mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'

import CreateNewLayoutModal from './index'

configure({ adapter: new Adapter() })

describe('CreateNewLayout', () => {
    test('should render modal', () => {
        const component = mount((
            <CreateNewLayoutModal />
        ))

        expect(component).toMatchSnapshot()
    })

    test('submit empty input', () => {
        const component = mount((
                <CreateNewLayoutModal
                    onSubmit={() => {}}
                    onClose={() => {}}
                />
        ))

        component.find('.green').simulate('click')
        expect(component.find('.error').length).toBe(1) 
    })

    test('fill api key & api secret', () => {
        const component = mount((
            <CreateNewLayoutModal
                onSubmit={() => {}}
                onClose={() => {}}
            />
    ))
    component.setState({label: 'test_layout'})        
    component.find('.green').simulate('click')
    expect(component.find('.error').length).toBe(0) 
    })
})  