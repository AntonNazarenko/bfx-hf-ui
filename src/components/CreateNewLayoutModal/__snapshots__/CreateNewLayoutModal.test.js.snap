// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewLayout should render modal 1`] = `
<CreateNewLayoutModal>
  <Modal
    actions={
      <Button
        className=""
        green={true}
        label={
          Array [
            <i
              className="icon-plus"
            />,
            <p>
              Add Layout
            </p>,
          ]
        }
        onClick={[Function]}
      />
    }
    className="hfui-createnewlayoutmodal__wrapper"
    label="Add Layout"
  >
    <div
      className="hfui-modal__wrapper"
    >
      <div
        className="hfui-modal__content hfui-createnewlayoutmodal__wrapper"
        onClick={[Function]}
      >
        <div
          className="hfui-modal__header"
        >
          <p
            className="hfui-modal__label"
          >
            Add Layout
          </p>
          <i
            className="icon-cancel"
          />
        </div>
        <Input
          onChange={[Function]}
          placeholder="Layout Name"
          type="text"
          value=""
        >
          <div
            className="hfui-input"
          >
            <input
              onChange={[Function]}
              placeholder="Layout Name"
              type="text"
              value=""
            />
          </div>
        </Input>
        <div
          className="hfui-modal__actions"
        >
          <Button
            className=""
            green={true}
            label={
              Array [
                <i
                  className="icon-plus"
                />,
                <p>
                  Add Layout
                </p>,
              ]
            }
            onClick={[Function]}
          >
            <button
              className="hfui-button green"
              onClick={[Function]}
              type="button"
            >
              <i
                className="icon-plus"
                key="icon"
              />
              <p
                key="text"
              >
                Add Layout
              </p>
            </button>
          </Button>
        </div>
      </div>
    </div>
  </Modal>
</CreateNewLayoutModal>
`;
